### PR TITLE
Enable setting SSL_CERT_FLAG_TLS_STRICT with ssl config

### PR DESCRIPTION
**Pull request by t8m**
This can be useful for ssl_new testcases.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: doc/man3/SSL_CONF_cmd.pod
Comment (tmshort) [inline doc/man3/SSL_CONF_cmd.pod:537]: Just noting that this flag is the equivalent to `-strict` on the s_client/s_server command line, but none of the command-line options reference config options (just the flags).

File: ssl/ssl_conf.c
Comment (tmshort) [inline ssl/ssl_conf.c:397]: Also noting that this is the first time `SSL_FLAG_TBL_CERT` is being used, it had been previously defined but never used.

**Review by tmshort — APPROVED** [review]
LGTM (pending CI; which is currently just Windows)

**Comment by t8m** [comment]
ping for second review

File: test/ssl-tests/04-client_auth.cnf.in
Comment (mattcaswell) [inline test/ssl-tests/04-client_auth.cnf.in:204]: I am slightly confused by this. We are expecting a *server* fail due to StrictCertCheck....but the option is set *client* side? Why does the server fail because of a client side option?

Comment (t8m) [reply test/ssl-tests/04-client_auth.cnf.in:204]: Because the client certificate as configured at the client will not be used by the client because it is rejected by the StrictCertCheck. As the issuer DN is not on the list sent by the server. The server will fail because it is configured to require a client cert and none is provided.

**Comment by t8m** [comment]
Ping again for the second review.

**Review by paulidale — APPROVED** [review]

**Pull request closed without merging**
